scrapers/utils.py

```diff
@@ -97,3 +98,8 @@ def get_md5hash(filepath):
             md5.update(data)
 
     return md5.hexdigest()
+
+def is_pdf(url):
+    response = requests.head(url)
+    content_type = response.headers.get('Content-Type', '')
+    return 'application/pdf' in content_type
```

Comment:
2 things:

1. Instead of breaking up into 2 checks, I would probably combine the check for null value here, with the _option_ to also check if it is a PDF as well (because I don't think we should assume it would be PDF for all circulars)

2. Please make sure there is an empty line at the bottom of a python file

```suggestion
def validate_and_get_download_link(base_url, download_link, check_content_pdf=True):
    # Add your check for null here, and raise if it is invalid
    is_valid_link = .....
    absolute_url = f"{base_url}/{download_link}"
    if check_content_pdf:
      response = requests.head(absolute_url)
      content_type = response.headers.get('Content-Type', '')
      is_pdf = 'application/pdf' in content_type
      if not is_pdf:
          raise ....
    return absolute_url
```
Then in scrapers, you can do:

```
validate_and_get_download_link(BASE_URL, link)
```